test/dataframes.jl

```diff
@@ -123,6 +123,56 @@ end
     @test d ≅ @transform(df, n1 = :i .* :g, n2 = :i .* :g)
 end
 
+@testset "@transform with @byrow" begin
+    df = DataFrame(
+        g = [1, 1, 1, 2, 2],
+        i = 1:5,
+        t = ["a", "b", "c", "c", "e"],
+        y = [:v, :w, :x, :y, :z],
+        c = [:g, :quote, :body, :transform, missing])
+
+    @test @transform(df, n = @byrow :i).n == df.i
```

Comment:
Maybe add some tests checking the whole output `DataFrame` not only its specific columns.
Also do we support `GroupedDataFrame` here?

Reply:
I have updated tests to be more reflective of the types of cases we are concerned with. 

I will updated the `GroupedDataFrame` tests once we think these are good. They are in a different file.